### PR TITLE
Terrain Surface Gradient List component.

### PR DESCRIPTION
**Pull request by sphrose**
Signed-off-by: sphrose <82213493+sphrose@users.noreply.github.com>

### CODE REVIEW

**Comment by sptramer** [comment]
@sphrose Please resolve the conflicts and then this PR can be reviewed.

File: content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+[Terrain Gem](/docs/user-guide/gems/reference/environment/terrain)
+
+## Properties ##
+
+| Property | Description | Values | Default |
+|-|-|-|-|
+| **GradientEntity** | The gradient entity to assign to this layer. | Gradient Entity | None |
+| **Surface Tag** | Sets the surface type that this gradient repersents. | Surface Tag | Mone |
+
+
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md:24]: As mentioned earlier, the provided link to the layer_spawner appears to be incorrect.
```suggestion

## Usage
1. Select a gradient by dragging an entity with a **Gradient** component to the **GradientEntity** field, or by using the select button.

2. Select the surface type that this gradient represents, by using the **Surface Tag** option.

3. Configure the dimensions and priority of the layer using the required [Terrain Layer Spawner](/docs/user-guide/components/reference/terrain/layer_spawner)

```

File: content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+Retrieves all the surfaces assigned to this component, along with the weight values which come from the gradient as the given position.
+
+**Parameters**  
+InPosition - Position to retrieve information for.
+Type: Vector3  
+SurfaceWeights \[out\] - A list of all the retrieved surfaces and weight values. 
+Type: SurfaceTagWeightMap
+
+**Return**
+None
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md:40]: - Reformat into a table to make it more clean. 
```suggestion
#### Parameters
| Parameter | Description | Type | 
| - | - | - |   
| InPosition | Position to retrieve information for. | Vector3 |
| SurfaceWeights \[out\] | A list of all the retrieved surfaces and weight values. | Type: SurfaceTagWeightMap |

#### Return
None

```

File: content/docs/user-guide/gems/reference/terrain/_index.md [hunk 3]
```diff
@@ -24,3 +24,9 @@ To enable the Terrain Gem, do the following:
 
 1. Use **O3DE Project Manager** or the command line to add the Terrain Gem to your project.
 2. Build your project using Project Manager, Visual Studio, or CMake.
+
+## Components
+
+| Component | Description | 
+| - | - |
+| [Terrain Surface Gradient List](/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md) | Allows you to define mappings between gradients and a surface types on a terrain layer. |
```
Comment (chanmosq) [inline content/docs/user-guide/gems/reference/terrain/_index.md:32]:
```suggestion
| [Terrain Surface Gradient List](/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md) | Define mappings between a gradient and a surface type on a terrain layer. |
```

**Review by chanmosq — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/components/reference/_index.md [hunk 4]
```diff
@@ -207,6 +207,11 @@ The components below are grouped by type as they appear in the O3DE Editor.
 | PhysX Collider Surface Tag Emitter | Enables a physics collider to emit surface tags. |
 | Shape Surface Tag Emitter | Enables a shape to emit surface tags. |
 
+### Terrain  
+
+| Component | Description | 
+| - | - |
+| [Terrain Surface Gradient List](/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md) | Allows you to define mappings between gradients and a surface types on a terrain layer. |
```
[the remark this answers is not in the document]
Comment (sphrose) [reply content/docs/user-guide/components/reference/_index.md:214]: Changed

File: content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+---
+title: Terrain Surface Gradient List component
+linktitle: TerrainSurfaceGradientList
+description: ' Open 3D Engine (O3DE) Terrain Surface Gradient List reference. '
+weight: 100
+---
+
+The **Terrain Surface Gradient List** component allows you to define mappings between a gradient and a surface on a terrain layer.
```
[the remark this answers is not in the document]
Comment (sphrose) [reply content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md:8]: Changed.

File: content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+You select a gradient either by dragging a entity containing a gradient component to the **GradientEntity** field, or by using the select button.
+Once a gradient is assigned, you can select the surface type that this gradient represents, by using the **Surface Tag** option.
+You can configure the dimensions and priority of the layer using the required [Terrain Layer Spawner](/docs/user-guide/components/reference/terrain/layer_spawner)
```
[the remark this answers is not in the document]
Comment (sphrose) [reply content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md:11]: This was defined in another PR which has now been merged, link should now exist.

File: content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md (hunk 1 above)
Comment (sphrose) [reply content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md:24]: This works now.

File: content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md (hunk 2 above)
Comment (sphrose) [reply content/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md:40]: Done, but now it doesn't match the other terrain components.

File: content/docs/user-guide/gems/reference/terrain/_index.md (hunk 3 above)
Comment (sphrose) [reply content/docs/user-guide/gems/reference/terrain/_index.md:32]: Changed.

**Review by sptramer — CHANGES_REQUESTED** [review]
Some minor link changes and file renames requested.

File: content/docs/user-guide/components/reference/_index.md [hunk 7]
```diff
@@ -213,6 +213,7 @@ The components below are grouped by type as they appear in the O3DE Editor.
 | - | - |
 | [Terrain Layer Spawner](/docs/user-guide/components/reference/terrain/layer_spawner/) | Spawns a terrain region contained within configurable bounds, and allows prioritization of overlapping terrain layers. |
 | [Terrain Height Gradient List](/docs/user-guide/components/reference/terrain/terrain_height_gradient_list.md) | Provides terrain height data from a list of gradients. |
+| [Terrain Surface Gradient List](/docs/user-guide/components/reference/terrain/terrain_surface_gradient_list.md) | Defines mappings between a gradient and a surface type on a terrain layer. |
```
[the remark this answers is not in the document]
Comment (sphrose) [reply content/docs/user-guide/components/reference/_index.md]: Changed.

File: content/docs/user-guide/components/reference/terrain/surface_gradient_list.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+### GetSurfaceWeights 
+
+Retrieves all the surfaces assigned to this component, along with the weight values which come from the gradient at the given position.
+
+**Parameters**  
+| Parameter | Description | Type | Return |
+| - | - | - | - |
+| InPosition | Position to retrieve information for. | Vector3 | None |
+| SurfaceWeights \[out\] | A list of all the retrieved surfaces and weight values. | SurfaceTagWeightMap | None |
+
```
[the remark this answers is not in the document]
Comment (sphrose) [reply content/docs/user-guide/components/reference/terrain/surface_gradient_list.md:43]: Changed.

File: content/docs/user-guide/components/reference/terrain/surface_gradient_list.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+| **Surface Tag** | Sets the surface type that this gradient represents. | Surface Tag | None |
+
+
+## TerrainAreaSurfaceRequestBus  ##
+
+Use the following request functions with the `TerrainAreaSurfaceRequestBus ` EBus interface to communicate with other components of your game.
+
+### GetSurfaceWeights 
+
+Retrieves all the surfaces assigned to this component, along with the weight values which come from the gradient at the given position.
```
[the remark this answers is not in the document]
Comment (sphrose) [reply content/docs/user-guide/components/reference/terrain/surface_gradient_list.md:36]: Changed, with minor tweaks.

**Review by sptramer — APPROVED** [review]

**Pull request merged**